plugin/src/init.server.lua

```diff
@@ -139,35 +143,44 @@ local function sendFullDMInfo(isSilent: boolean?)
 		warn(`[Luau Language Server] Sending full DM info failed: {result.StatusCode}: {result.Body}`)
 		connected.Value = false
 	else
-		if not isSilent then
-			print("[Luau Language Server] Listening for DataModel changes")
-		end
 		connected.Value = true
 	end
 end
 
 local function watchChanges(isSilent)
+	local wasUpdated = false
 	if connected.Value or Settings == nil then
 		if not isSilent then
 			warn("[Luau Language Server] Connecting to server failed: invalid settings")
 		end
 		return
 	end
 	cleanup()
+	local batchLoop = task.spawn(function()
+		while true do
+			task.wait(0.5)
+			if wasUpdated then
+				sendFullDMInfo()
+				wasUpdated = false
+			end
+		end
+	end)
```

Comment:
Maybe instead of busy-waiting we can be a bit more clever about it and do it on demand?

Something along the lines of

```lua
local function deferSend()
  if sendTask then
    task.cancel(sendTask)
  end
  sendTask = task.delay(0.5, function()
    sendTask = nil
    ...
  end)
end
```